File: irrgarten/GameState.java
package irrgarten;

public class GameState {
    private final String labyrinth;
    private final String players;
    private final String monsters;
    private final int currentPlayer;
    private final boolean winner;
    private final String log;

    public GameState(String labyrinth, String players, String monsters, int currentPlayer, boolean winner, String log) {
        this.labyrinth = labyrinth;
        this.players = players;
        this.monsters = monsters;
        this.currentPlayer = currentPlayer;
        this.winner = winner;
        this.log = log;
    }

    public String getLabyrinth() {
        return labyrinth;
    }

    public String getPlayers() {
        return players;
    }

    public String getMonsters() {
        return monsters;
    }

    public int getCurrentPlayer() {
        return currentPlayer;
    }

    public boolean isWinner() {
        return winner;
    }

    public String getLog() {
        return log;
    }

    public String toString(){
        String aux = "";

        aux += getLabyrinth();
        aux += "\n";
        aux += getPlayers();
        aux += "\n";
        aux += getMonsters();
        aux += "\n";
        aux += getCurrentPlayer();
        aux += "\n";
        aux += isWinner();
        aux += "\n";
        aux += getLog();
        aux += "\n";

        return aux;
    }
}

File: irrgarten/GameState.rb
module Irrgarten
    class GameState
        #Constructor  de la clase con parámetros.
        def initialize(labytinthv, players, monsters, currentplayer, winner, log )
          # Set the instance variables within the constructor
          @labyrinthv = labytinthv
          @players = players
          @monsters = monsters
          @currentPlayer = currentplayer
          @winner = winner
          @log = log
        end
        #private
        #Get para Labyrinthv.
        def getLabyrinthv
            @labyrinthv
        end
        #Get para players.
        def getplayers
            @players
        end
        #Get para monsters.
        def getmonsters
            @monsters
        end
        #Get para currentPlayer.
        def getcurrentPlayer
            @currentPlayer
        end
        #Get para winner.
        def getWinner
            @winner
        end
        #Get para Log.
        def getLog
            @log
        end
        
    end
end

